tests/test.py

```diff
@@ -7246,6 +7249,117 @@ def testRotatedKeysAcraBlock(self):
         self.assertEqual(rows[0]['searchable_acrablock'][33:33+4], encrypted_term[:4])
 
 
+class TestSearchableTransparentEncryptionWithJOINs(BaseSearchableTransparentEncryption):
+    encryptor_table_join = sa.Table(
+        'test_searchable_transparent_encryption_join', metadata,
+        sa.Column('id', sa.Integer, primary_key=True),
+        sa.Column('specified_client_id',
+                  sa.LargeBinary(length=COLUMN_DATA_SIZE)),
+        sa.Column('default_client_id',
+                  sa.LargeBinary(length=COLUMN_DATA_SIZE)),
+
+        sa.Column('number', sa.Integer),
+        sa.Column('zone_id', sa.LargeBinary(length=COLUMN_DATA_SIZE)),
+        sa.Column('raw_data', sa.LargeBinary(length=COLUMN_DATA_SIZE)),
+        sa.Column('nullable', sa.Text, nullable=True),
+        sa.Column('searchable', sa.LargeBinary(length=COLUMN_DATA_SIZE)),
+        sa.Column('searchable_acrablock', sa.LargeBinary(length=COLUMN_DATA_SIZE)),
+        sa.Column('empty', sa.LargeBinary(length=COLUMN_DATA_SIZE), nullable=False, default=b''),
+        sa.Column('token_i32', sa.Integer(), nullable=False, default=1),
+        sa.Column('token_i64', sa.BigInteger(), nullable=False, default=1),
+        sa.Column('token_str', sa.Text, nullable=False, default=''),
+        sa.Column('token_bytes', sa.LargeBinary(length=COLUMN_DATA_SIZE), nullable=False, default=b''),
+        sa.Column('token_email', sa.Text, nullable=False, default=''),
+    )
+
+    def setUp(self):
+        super().setUp()
+        self.engine1.execute(self.encryptor_table_join.delete())
+
+    def testSearchWithJoinedTable(self):
+        context = self.get_context_data()
+        search_term = context['searchable']
+
+        # Insert searchable data and some additional different rows
+        self.insertRow(context)
+        self.insertDifferentRows(context, count=5)
+
+        # Insert the same data into encryptor_table_join table
+        self.insertRow(context, query=self.encryptor_table_join.insert())
+        self.insertDifferentRows(context, count=5, query=self.encryptor_table_join.insert())
+
+        rows = self.executeSelect2(
+            sa.select(
+                self.encryptor_table.c.number,
+                self.encryptor_table_join.c.searchable,
+                self.encryptor_table_join.c.default_client_id,
+                self.encryptor_table_join.c.raw_data,
+                self.encryptor_table_join.c.specified_client_id,
+                self.encryptor_table_join.c.zone_id,
+                self.encryptor_table_join.c.empty,
+            ).
+            where(self.encryptor_table.c.searchable == sa.bindparam('searchable')).
+            join(self.encryptor_table_join, self.encryptor_table_join.c.searchable==sa.bindparam('searchable')),
```

Comment:
can we place `where` below `join` expression? because it is unusual to read sqlalchemy code written in different order than real SQL )

Reply:
Sure!